go/interpreter/lfvm/instructions.go

```diff
@@ -653,8 +654,10 @@ func opGasPrice(c *context) {
 func opBalance(c *context) error {
 	slot := c.stack.peek()
 	address := tosca.Address(slot.Bytes20())
-	if err := gasEip2929AccountCheck(c, address); err != nil {
-		return err
+	if c.isAtLeast(tosca.R09_Berlin) {
+		if err := c.useGas(getAccessCost(c.context.AccessAccount(address))); err != nil {
```

Comment:
Maybe for a follow-up: Since these two checks always come in pairs, maybe the revision check should move inside the `getAccessCost` function. In pre-Berlin cases, a cost of `0` would then be returned.